--- FreeBSD.cpp-orig	2017-11-29 02:25:12.000000000 -0500
+++ FreeBSD.cpp	2018-05-15 11:47:46.954027786 -0400
@@ -320,8 +320,11 @@
        Triple.getArch() == llvm::Triple::ppc) &&
       D.getVFS().exists(getDriver().SysRoot + "/usr/lib32/crt1.o"))
     getFilePaths().push_back(getDriver().SysRoot + "/usr/lib32");
-  else
+  else {
+    //- KEWB
+    getFilePaths().push_back(getDriver().Dir + "/../lib");
     getFilePaths().push_back(getDriver().SysRoot + "/usr/lib");
+  }
 }
 
 ToolChain::CXXStdlibType FreeBSD::GetDefaultCXXStdlibType() const {
