--- FreeBSD.cpp-orig	2018-10-10 09:27:25.000000000 -0400
+++ FreeBSD.cpp	2020-03-26 09:55:59.740275394 -0400
@@ -347,8 +347,11 @@
        Triple.getArch() == llvm::Triple::ppc) &&
       D.getVFS().exists(getDriver().SysRoot + "/usr/lib32/crt1.o"))
     getFilePaths().push_back(getDriver().SysRoot + "/usr/lib32");
-  else
+  else {
+    //- KEWB
+    getFilePaths().push_back(getDriver().Dir + "/../lib");
     getFilePaths().push_back(getDriver().SysRoot + "/usr/lib");
+  }
 }
 
 ToolChain::CXXStdlibType FreeBSD::GetDefaultCXXStdlibType() const {
