--- HandleLibCXXABI.cmake-orig	2018-11-16 12:22:22.454160382 -0500
+++ HandleLibCXXABI.cmake	2018-11-16 12:42:42.696093248 -0500
@@ -41,7 +41,7 @@
         get_filename_component(ifile ${fpath} NAME)
         set(src ${incpath}/${fpath})
 
-        set(dst ${LIBCXX_BINARY_INCLUDE_DIR}/${dstdir}/${fpath})
+        set(dst ${LIBCXX_BINARY_INCLUDE_DIR}/${dstdir}/${ifile})
         add_custom_command(OUTPUT ${dst}
             DEPENDS ${src}
             COMMAND ${CMAKE_COMMAND} -E copy_if_different ${src} ${dst}
@@ -49,7 +49,7 @@
         list(APPEND abilib_headers "${dst}")
 
         if (NOT LIBCXX_USING_INSTALLED_LLVM AND LIBCXX_HEADER_DIR)
-          set(dst "${LIBCXX_HEADER_DIR}/include/c++/v1/${dstdir}/${fpath}")
+          set(dst "${LIBCXX_HEADER_DIR}/include/c++/v1/${dstdir}/${ifile}")
           add_custom_command(OUTPUT ${dst}
               DEPENDS ${src}
               COMMAND ${CMAKE_COMMAND} -E copy_if_different ${src} ${dst}
