--- testit-orig	2015-11-01 16:14:22.000000000 -0500
+++ testit	2017-01-07 14:51:53.193933312 -0500
@@ -28,7 +28,7 @@

 if [ -z "$CC" ]
 then
-	if which xcrun >/dev/null
+	if which xcrun > /dev/null 2>&1
 	then
 		CC="xcrun clang++"
 	else
@@ -38,7 +38,7 @@

 if [ -z "$CXX_LANG" ]
 then
-    CXX_LANG=c++11
+    CXX_LANG=c++1y
 fi
 if [ -z "$OPTIONS" ]
 then
@@ -56,6 +56,11 @@
        SOURCE_LIB="-L$LIBCXX_ROOT/lib"
 fi

+if [ -z "$LIBS" ]
+then
+	LIBS="-lpthread"
+fi
+
 case $TRIPLE in
   *-*-mingw* | *-*-cygwin* | *-*-win*)
 	TEST_EXE=test.exe
